patch sequence page (#1205)
ocaml/v3.ocaml.org: data/tutorials/lg_07_objects.md

  ```

Would this modify the private internal representation of my `container`
class, by prepending `x` to the list of widgets? No it wouldn't. If you run 
the above code, you'll see it throws an error:

```ocaml
Error: Unbound value container
```

Yet, the private variable `widgets` would be unaffected by this or any other
change attempted by the outside code. This means, for example, that you
could change the internal representation to use an array at some later
date, and no code outside the class would need to be changed.

Last, but not least, we have implemented the previously virtual
`repaint` function so that `container#repaint` will repaint all of the
contained widgets. Notice the use of `List.iter` to iterate over the list,

> and I also use an anonymous function expression that might be unfamiliar:

Reviewer: ```suggestion
and also the use of an anonymous function expression that might be unfamiliar:
```

```ocaml
# (fun w -> w#repaint);;
- : < repaint : 'a; .. > -> 'a = <fun>
```
This defines an anonymous function with one argument `w` that just
calls `w#repaint` (the `repaint` method on widget `w`).

In this instance, our `button` class is simple (rather unrealistically

> simple in fact, but nevermind that):

Reviewer: If it's unrealistically simple, does it benefit the reader? If so, does mentioning it's unrealistically simple downplay its importance?

Notes:

1. This function has an optional argument ([see the previous chapter](https://ocaml.org/docs/functors))
 which is used to pass in the optional callback function. The
 callback is called when the button is pressed.
1. The expression `inherit container name as super` names the
 superclass `super`. I use this in the `repaint` method:
 `super#repaint`. This expressly calls the superclass method.
1. Pressing the button (calling `button#press` in this rather
 simplistic code) sets the button to `Pressed` and calls
 the callback function, if one was defined. Notice that the
 `callback` variable is either `None` or `Some f`, meaning it
 has type `(unit -> unit) option`. Reread the [previous chapter](https://ocaml.org/docs/functors) if you
 are unsure about this.
1. Notice a strange thing about the `callback` variable. It's defined
 as an argument to the class, but any method can see and use it. In
 other words, the variable is supplied when the object is

>  constructed, and it also persists over the lifetime of the object.

Reviewer: ```suggestion
 constructed, and it also persists over the object's lifetime.
```

# let l = new label "label" "Press me!";;
val l : label = <obj>
# [b; l];;
Error: This expression has type label but an expression was expected of type
         button
       The first object type has no method add
```
We created a button `b` and a label `l` and then tried to create a list

> containing both, but we got an error. Yet `b` and `l` are both `widget`s,

Reviewer: ```suggestion
containing both, but it threw an error. Yet `b` and `l` are both `widget`s,
```

`=` and `<>` can be used to compare objects for *physical* equality (an
object and its copy are not physically identical). You can also use `<`
etc., which provides an ordering of objects based apparently on their
IDs.

## Objects Without Class
Here we examine how to use objects pretty much like records, without
necessarily using classes.

### Immediate Objects and Object Types
Objects can be used instead of records. Plus, they have some nice properties
that can make them preferable to records in some cases. We saw that the
canonical way of creating objects is to first define a class, then use
this class to create individual objects. This can be cumbersome in some

> situations becaus class definitions are more than a type definition and

Reviewer: ```suggestion
situations because class definitions are more than a type definition and
```
